### PR TITLE
Performance and size optimizations

### PR DESCRIPTION
**Pull request by maciejhirsz**
Now using [Pixelate](https://crates.io/crates/pixelate). This changes the output files from full RGBA to a palette based images.

+ For `classic` the palette is 6 bytes (2 colors) and the pre-compression bitmap is 1 bit per pixel.
+ For `ethereum` the palette is 9 bytes (3 colors) and the pre-compression bitmap is 2 bits per pixel.

The end result is that generating images is ~10x faster and the produced images are ~4x smaller.

### CODE REVIEW

File: src/classic.rs [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
+use std::io;
 use hsl::HSL;
-use {hsl_to_rgba, fill_rect, rgba};
+use pixelate::{Color, Image, Error};
 
-fn white() -> Rgba<u8> {
-	rgba(255, 255, 255, 255)
+use crate::util::{create_image_data, hsl_to_rgb};
+
+pub struct Classic {
```
Comment (tomusdrw) [inline src/classic.rs:23]: Would be good to add docs to public items (but since they were not here earlier it's a separate issue)

**Review by tomusdrw — APPROVED** [review]
Nice, the size optimizations look impressive! :tada:

File: src/classic.rs (hunk 1 above)
Comment (maciejhirsz) [reply src/classic.rs:23]: Added doc comments to both structs.

**Review by tomusdrw — APPROVED** [review]
lgtm, but I guess only @debris is able to merge this

**Review by debris — APPROVED** [review]

**Comment by debris** [comment]
added both of you as collaborators

**Pull request merged**